webapp/app/forms/steps/eligibility_steps.py

```diff
@@ -52,37 +62,72 @@ class IncomesForm(SteuerlotseBaseForm):
                                   'text': _l('form.eligibility.elster-account-help')}})
 
     def __init__(self, **kwargs):
-        super(EligibilityStepIncomes, self).__init__(
+        super(EligibilityIncomesFormSteuerlotseStep, self).__init__(
             title=_('form.eligibility.income-title'),
             intro=_('form.eligibility.income-intro'),
             form=self.IncomesForm,
             **kwargs,
             header_title=_('form.eligibility.header-title'),
             template='eligibility/form_incomes.html')
 
+    def _main_handle(self, stored_data):
+        super()._main_handle(stored_data)
+        self.render_info.additional_info['next_button_label'] = _('form.eligibility.send-button')
+        return stored_data
 
-class EligibilityStepResult(DisplayStep):
+
+class EligibilityResultFormSteuerlotseStep(DisplaySteuerlotseStep):
     name = 'result'
 
     def __init__(self, **kwargs):
-        kwargs['prev_step'] = EligibilityStepIncomes
-        super(EligibilityStepResult, self).__init__(title=_('form.eligibility.result-title'), **kwargs)
+        kwargs['prev_step'] = EligibilityIncomesFormSteuerlotseStep
+        super(EligibilityResultFormSteuerlotseStep, self).__init__(title=_('form.eligibility.result-title'), **kwargs)
+
+    def _main_handle(self, stored_data):
+        super()._main_handle(stored_data)
+        eligibility_result = self._validate_eligibility(stored_data)
+        self.render_info.additional_info['eligibility_errors'] = eligibility_result
+        return stored_data
 
-    def render(self, data, render_info):
-        eligibility_errors = render_info.additional_info.get('eligibility_errors')
+    @staticmethod
+    def _validate_eligibility(stored_data):
+        """
+        Method to find find out whether a user is eligible to use the Steuerlotse and give errors for this result.
+
+        :return: eligibility_result
+        :rtype: EligibilityResult
+        """
+        errors = []
+        missing_inputs = []
+        try:
+            ExpectedEligibility.parse_obj(stored_data)
+        except ValidationError as e:
+            errors = [raw_e.exc.message for raw_e in e.raw_errors if isinstance(raw_e.exc, InvalidEligiblityError)]
+            missing_inputs = [raw_e.exc for raw_e in e.raw_errors if isinstance(raw_e.exc, MissingError)]
+        if not errors and missing_inputs:
+            raise IncorrectEligibilityData
+        return errors
+
+    def render(self):
+        eligibility_errors = self.render_info.additional_info.get('eligibility_errors')
```

Comment:
I am with you on this one and I think we can change it. Nevertheless, we will have to change that again for the next changes to the eligibility flow. So, it might be not that important